tokio-util/src/sync/cancellation_token.rs

```diff
@@ -99,6 +99,19 @@ pin_project! {
     }
 }
 
+pin_project! {
+    /// A Future that is resolved once the corresponding [`CancellationToken`]
+    /// is cancelled or a given Future gets resolved. It is biased towards the
+    /// Future completion.
+    #[must_use = "futures do nothing unless polled"]
+    struct RunUntilCancelledFuture<C, F: Future> {
```

Comment:
Know this type basically is a biased select future I think because `C` technically is not required to be some sort of `CancellationToken`. I think we could replace the usage of this type with a `tokio::select!` if we add the macros feature to `tokio-util`s `tokio` dependency. @Darksonn do you have a optionion on that?